pkg/util/cmdutil/console.go

```diff
@@ -50,7 +50,11 @@ func Interactive() bool {
 
 // InteractiveTerminal returns true if the current terminal session is interactive.
 func InteractiveTerminal() bool {
-	return terminal.IsTerminal(int(os.Stdin.Fd()))
+	// if we're piping in stdin, we're clearly not interactive, as there's no way for a user to
+	// provide input.  If we're piping stdout, we also can't be interactive as there's no way for
+	// users to see prompts to interact with them.
+	return terminal.IsTerminal(int(os.Stdin.Fd())) &&
+		terminal.IsTerminal(int(os.Stdout.Fd()))
```

Comment:
this is a change in behavior... but i think it's desirable.  Specifically, if you did:  `pulumi up > out.txt` we'd basically just hang.  That's because we'd get to the part about asking if you wanted to proceed, and you'd never see it.  AFAICT, this behavior is much more sensible.  but i want unix aficionados to weigh in here.

Reply:
This sounds correct to me.